src/datastore/postgres-store.ts

```diff
@@ -3644,11 +3664,30 @@ export class PgDataStore
       text: insertQuery,
       values,
     };
+    // zonefile insertion query
+    const zonefileInsertQuery = `INSERT INTO zonefiles (zonefile, zonefile_hash) VALUES ${zonefileInsertParams}`;
+    const insertZonefileQueryName = `insert-batch-zonefiles_${columnCount}x${subdomains.length}`;
+    const insertZonefilesEventQuery: QueryConfig = {
+      name: insertZonefileQueryName,
+      text: zonefileInsertQuery,
+      values: zonefileValues,
+    };
     try {
-      const res = await client.query(insertBnsSubdomainsEventQuery);
-      if (res.rowCount !== subdomains.length) {
-        throw new Error(`Expected ${subdomains.length} inserts, got ${res.rowCount}`);
-      }
+      const result = await this.queryTx(async client => {
```

Comment:
This should use the existing `client` passed to the function rather than opening a new sql tx.